packages/api-server/src/rate-limit.ts

```diff
@@ -88,6 +101,35 @@ export async function rateLimit(
   return isBan;
 }
 
+export async function _rateLimit(
+  rpcMethod: string,
+  reqId: string
+): Promise<{ error: JSONRPCError; remainSecs: number } | undefined> {
+  const isExist = await accessGuard.isExist(rpcMethod, reqId);
+  if (!isExist) {
+    await accessGuard.add(rpcMethod, reqId);
+  }
+
+  const isOverRate = await accessGuard.isOverRate(rpcMethod, reqId);
+  if (isOverRate) {
+    const remainSecs = await accessGuard.getKeyTTL(rpcMethod, reqId);
+
+    const message = `Too Many Requests, IP: ${reqId}, please wait ${remainSecs}s and retry. RPC method: ${rpcMethod}.`;
+    const error: JSONRPCError = {
+      code: LIMIT_EXCEEDED,
+      message: message,
+    };
+
+    logger.debug(
+      `Rate Limit Exceed, ip: ${reqId}, method: ${rpcMethod}, ttl: ${remainSecs}s`
+    );
+    return { error, remainSecs };
```

Comment:
What about throwing the error?

Reply:
throwing requires some refactor on http/ws, maybe leave it to future PRs